lib/puppet/functions/dnsquery/a.rb

```diff
@@ -3,16 +3,23 @@
 # Retrieves DNS A records for a domain and returns them as an array.
 Puppet::Functions.create_function(:'dnsquery::a') do
   # @param domain the dns domain to lookup
+  # @param config_info used to override the config for Resolve::DNS.new
   # @param block an optional lambda to return a default value in case the lookup fails
   # @return An array of A answers matching domain
   dispatch :dns_a do
     param 'Stdlib::Fqdn', :domain
+    optional_param 'Dnsquery::Config_info', :config_info
     optional_block_param :block
     return_type 'Array[Stdlib::IP::Address::V4::Nosubnet]'
   end
 
-  def dns_a(domain)
-    ret = Resolv::DNS.new.getresources(
+  def dns_a(domain, config_info = nil)
+    resolver = if config_info
+                 Resolv::DNS.new(config_info.transform_keys(&:to_sym))
+               else
+                 Resolv::DNS.new
+               end
```

Comment:
This is repeated over and over.  Maybe we can add a `util.rb` file, require_relative it and extract this code in a function to avoid code duplication?